go-controller/pkg/ovn/gateway_init.go

```diff
@@ -72,7 +73,15 @@ func (oc *DefaultNetworkController) cleanupStalePodSNATs(nodeName string, nodeIP
 			}
 		}
 		podIPs, err := util.GetPodIPsOfNetwork(&pod, oc.NetInfo)
-		if err != nil {
+		if err != nil && errors.Is(err, util.ErrNoPodIPFound) {
```

Comment:
@tssurya just a small nit; I'd rather have:

```
if err != nil {
    if errors.Is(err, util.ErrNoPodIPFound) {
        klog.Warningf("Unable to fetch podIPs for pod %s/%s: %v", pod.Namespace, pod.Name, err)
        continue
    }
    return fmt.Errorf("unable to fetch podIPs for pod %s/%s: %w", pod.Namespace, pod.Name, err)
}
```

rather than two err != nil conditions.

Reply:
yeah I did think of this, but some reviewers don't like too much nesting haha hence went with that way..would you like me to change it this one? I already cherry-picked this downstream, hoping to not change commit-sha's ...